apps/ledger-live-desktop/src/renderer/modals/Stake/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useCallback, useMemo } from "react";
+import { useDispatch } from "react-redux";
+import { useTranslation } from "react-i18next";
+import { Icon, Text } from "@ledgerhq/react-ui";
+import { useRampCatalog } from "@ledgerhq/live-common/platform/providers/RampCatalogProvider/index";
+import { getAllSupportedCryptoCurrencyTickers } from "@ledgerhq/live-common/platform/providers/RampCatalogProvider/helpers";
+import { Account, AccountLike } from "~/../../../libs/ledgerjs/packages/types-live/lib";
+import { closeModal, openModal } from "~/renderer/actions/modals";
+import { useProviders } from "~/renderer/screens/exchange/Swap2/Form";
+import Modal, { ModalBody } from "~/renderer/components/Modal";
+import Box from "~/renderer/components/Box";
+import EntryButton from "~/renderer/components/EntryButton/EntryButton";
+import CoinsIcon from "./assets/CoinsIcon";
+import { useHistory } from "react-router-dom";
+
+interface StakeModalProps {
+  account: AccountLike;
+  parentAccount?: Account;
+}
+
+const StakeModal = ({ account, parentAccount }: StakeModalProps) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const rampCatalog = useRampCatalog();
+  const { providers, storedProviders } = useProviders();
+
+  const onRampAvailableTickers = useMemo(() => {
+    if (!rampCatalog.value) {
+      return [];
+    }
+    return getAllSupportedCryptoCurrencyTickers(rampCatalog.value.onRamp);
+  }, [rampCatalog.value]);
+
+  const swapAvailableIds =
+    providers || storedProviders
+      ? (providers || storedProviders)
+          .map(({ pairs }: any) => pairs.map(({ from, to }: any) => [from, to]))
```

Comment:
i think there's like 20,000 pairs, so I would suggest wrapping this in `useMemo`

Reply:
Whilst we're wrapping in `useMemo`, we probably should also memoise `availableOnSwap` too, as I assume `swapAvailableIds` could also be very large.

Reply:
Done!